lib/pub_sub/subscriber.rb

```diff
@@ -5,23 +5,55 @@ def self.subscribe
     end
 
     def subscribe
-      PubSub.config.subscriptions.keys.each do |service_identifier|
+      topics = PubSub.config.subscriptions.keys.map do |service_identifier|
         subscribe_to_service(service_identifier)
       end
+      set_queue_policy(topics.map(&:arn))
     end
 
+    private
+
     def subscribe_to_service(service_identifier)
       topic = Aws::SNS::Topic.new(sns.create_topic(name: service_identifier).topic_arn)
-      queue_data = Aws::SQS::Client.new.get_queue_attributes(queue_url: Queue.new.queue_url, attribute_names: ["QueueArn"])
-      queue_arn = queue_data.attributes["QueueArn"]
       topic.subscribe(endpoint: queue_arn, protocol: 'sqs')
       PubSub.logger.info "Subscribed #{queue_arn} to #{topic.arn}"
+      topic
     end
 
-    private
+    def set_queue_policy(topic_arns)
```

Comment:
But topic_arns is an array, which contains both topic 1 and topic 2, so these topics should both be included in the ArnEquals condition.  The policy of the queue is overwritten each time though, so if you had a custom policy with additional statements, that would be destroyed, however, that's what we want, otherwise we'd be adding another policy each time we run the subscription code.